supersetapiclient/base.py

```diff
@@ -307,39 +250,28 @@ def find(self, page_size: int = 100, page: int = 0, **kwargs):
 
     def count(self):
         """Count objects."""
-
-        # To do : add kwargs parameters for more flexibility
         response = self.client.get(self.base_url)
-
-        if response.status_code not in (200, 201):
-            logger.error(response.text)
         raise_for_status(response)
-
-        json = response.json()
-        return (json['count'])
+        return response.json()['count']
 
     def find_one(self, **kwargs):
         """Find only object or raise an Exception."""
         objects = self.find(**kwargs)
         if len(objects) == 0:
-            raise NotFound(f"No {self.base_object.__name__} has been found.")
+            raise NotFound(f"No {self.base_object.__name__} found")
+        if len(objects) > 1:
+            raise MultipleFound(f"Multiple {self.base_object.__name__} found")
         return objects[0]
 
     def add(self, obj) -> int:
-        """Create a object on remote."""
-
-        o = {}
-        for c in self.add_columns:
-            if hasattr(obj, c):
-                value = getattr(obj, c)
-
-                if c in obj.JSON_FIELDS:
-                    value = json.dumps(value)
-                o[c] = value
+        """Create an object on remote."""
 
+        o = obj.to_json(columns=self.add_columns)
         response = self.client.post(self.base_url, json=o)
         raise_for_status(response)
-        return response.json().get("id")
+        obj.id = response.json().get("id")
+        obj._parent = self
+        return obj.id
 
     def export(self, ids: List[int], path: Union[Path, str]) -> None:
```

Comment:
I am not sure why this method should be attach to ObjectFactories and not directly to Object

Reply:
The API supports exporting multiple items in one request. I thought we might as well keep that support. `Object.export()` is just a special case of that, exporting a single item.